### PR TITLE
[DRAFT] [DO NOT MERGE] Entities support prototype

### PR DESCRIPTION
**Pull request by dmitryax**
Updated version of https://github.com/tigrannajaryan/opentelemetry-collector/pull/4

Additional changes:
- Resource now has additional slice fields to reference attributes for the entity that this resource represents.
- EntitiesData now includes a resource that represents a parent entity. It allows all entities in the EntitiesData message to have local identities in the context of the parent entity.
- Full support of `entities` pipeline
- Components with entities support: 
  - `otlp` receiver
  - `memory_limiter` processor
  - `debug` exporter

### CODE REVIEW

File: pdata/ptrace/ptraceotlp/generated_exportpartialsuccess.go [hunk 1]
```diff
@@ -45,6 +45,14 @@ func (ms ExportPartialSuccess) MoveTo(dest ExportPartialSuccess) {
 	*ms.orig = otlpcollectortrace.ExportTracePartialSuccess{}
 }
 
+func (ms ExportPartialSuccess) getOrig() *otlpcollectortrace.ExportTracePartialSuccess {
+	return ms.orig
+}
+
+func (ms ExportPartialSuccess) getState() *internal.State {
+	return ms.state
+}
```
Comment (tigrannajaryan) [inline pdata/ptrace/ptraceotlp/generated_exportpartialsuccess.go:54]: There are a bunch of changes like this which seem unrelated to entities?

File: exporter/internal/otlptext/databuffer.go [hunk 2]
```diff
@@ -280,6 +280,21 @@ func (b *dataBuffer) logExemplars(description string, se pmetric.ExemplarSlice)
...
 
+func (b *dataBuffer) logResourceEntities(rers pcommon.ResourceEntityRefSlice) {
+	if rers.Len() == 0 {
+		return
+	}
+
+	b.logEntry("Entities:")
+	for i := 0; i < rers.Len(); i++ {
+		rer := rers.At(i)
+		b.logEntry("Entity Ref #%d", i)
```
[the remark this answers is not in the document]
Comment (dmitryax) [reply exporter/internal/otlptext/databuffer.go:291]: Sure. Let me add it

File: pdata/ptrace/ptraceotlp/generated_exportpartialsuccess.go (hunk 1 above)
Comment (dmitryax) [reply pdata/ptrace/ptraceotlp/generated_exportpartialsuccess.go:54]: They are semi-related. Just helper shortcuts to make Message->Slice work in the pcommon module (it wasn't supported before). I'll clean this up, just didn't want to block further prototyping.

Comment (dmitryax) [reply pdata/ptrace/ptraceotlp/generated_exportpartialsuccess.go]: done
